### PR TITLE
Improve coverage accountability

### PR DESCRIPTION
**Pull request by tbssajal**

### CODE REVIEW

File: autonity/solidity/test/accountability.js [hunk 1]
```diff
@@ -705,5 +705,82 @@ contract('Accountability', function (accounts) {
...
+        "block": 10,
+        "epoch": 0,
+        "reportingBlock": 11,
+        "messageHash": 0,
+      }
+      let balance = web3.utils.toWei("10", "ether");
+      await web3.eth.sendTransaction({from: validators[0].treasury, to: reporter, value: balance});
+      
+      await truffleAssert.fails(
+        accountability.handleEvent.call(event, {from: reporter}),
```
Comment (lorenzo-dev1) [inline autonity/solidity/test/accountability.js:763]: Can you add a comment explaining why we are using call here for future readers

File: autonity/solidity/test/accountability.js [hunk 2]
```diff
@@ -705,5 +705,82 @@ contract('Accountability', function (accounts) {
...
+      let eventCount = 3;
+      let currentProof = "0x";
+      for (let i = 0; i < eventCount; i++) {
+        let rawProof = [];
+        rawProof.push(i);
+        event.chunkId = i;
+        event.rawProof = rawProof;
+        let request = (await accountability.handleEvent.request(event, {from: reporter}));
+        let receipt = await utils.signAndSendTransaction(reporter, accountability.address, reporterPrivateKey, request);
+        assert.equal(receipt.status, true, "transcation failed");
```
Comment (lorenzo-dev1) [inline autonity/solidity/test/accountability.js:777]:
```suggestion
        assert.equal(receipt.status, true, "transaction failed");
```

File: autonity/solidity/test/accountability.js [hunk 3]
```diff
@@ -705,5 +705,82 @@ contract('Accountability', function (accounts) {
...
+        let currentEvent = await accountability.getReporterChunksMap({from: reporter});
+        let hexNumber = (i > 15) ? i.toString(16) : "0" + i.toString(16);
+        currentProof = currentProof + hexNumber;
+        assert.equal(currentEvent.rawProof, currentProof, "proof mismatch");
+      }
```
Comment (lorenzo-dev1) [inline autonity/solidity/test/accountability.js:782]: Can you add one more call to _handleEvent? it should revert with the proof verification failure. It is to check that the `readyToProcess` logic works

**Comment by tbssajal** [comment]
> Could you add a test case also for `require(_event.reporter == msg.sender, "event reporter must be caller");`

Added

File: autonity/solidity/test/accountability.js [hunk 4]
```diff
@@ -705,5 +710,265 @@ contract('Accountability', function (accounts) {
...
+        await truffleAssert.fails(
+          accountability.handleEvent.call(event, {from: reporter[i]}),
+          truffleAssert.ErrorType.REVERT,
+          "failed proof verification"
+        );
+      }
+
+    });
+
+    it.skip("has proper value in fields in storage", async function () {
```
Comment (lorenzo-dev1) [inline autonity/solidity/test/accountability.js:942]:
```suggestion
    it.skip("can start with chunkId = 1", async function () {
```

This name is more representative, unless I am missing something?

File: autonity/solidity/test/accountability.js [hunk 5]
```diff
@@ -705,5 +710,265 @@ contract('Accountability', function (accounts) {
...
+        "messageHash": 0,
+      };
+
+      let currentProof = "0x";
+      let eventCount = event.chunks - 1;
+      for (let chunkId = 0; chunkId < eventCount; chunkId++) {
+        let rawProof = [];
+        rawProof.push(chunkId);
+        event.chunkId = chunkId;
+        event.rawProof = rawProof;
```
Comment (lorenzo-dev1) [inline autonity/solidity/test/accountability.js:913]: would be better to have different rawproofs for the 2 different validators, so we are sure that there are no interferences

File: autonity/solidity/test/accountability.js (hunk 4 above)
Comment (lorenzo-dev1) [inline autonity/solidity/test/accountability.js:942]: what is the purpose of this test? Checking the start chunkid?

Comment (tbssajal) [reply autonity/solidity/test/accountability.js:942]: Yes, I will update the description. Depending on soln, we may need to delete the test.

**Review by lorenzo-dev1 — APPROVED** [review]
LGTM

**Pull request merged**